apps/prairielearn/assets/scripts/navbarClient.ts

```diff
@@ -0,0 +1,154 @@
+import { onDocumentReady } from '@prairielearn/browser-utils';
+import Cookies from 'js-cookie';
+
+const expires = 30; // in units of days
+
+onDocumentReady(() => {
+  const usernameNav = document.getElementById('username-nav');
+  const accessAsAdministrator = usernameNav.dataset.accessAsAdministrator === 'true';
+  const viewType = usernameNav.dataset.viewType;
+  const authnCourseRole = usernameNav.dataset.authnCourseRole;
+  const authnCourseInstanceRole = usernameNav.dataset.authnCourseInstanceRole;
+  // Note: this corresponds to `user_with_requested_uid_has_instructor_access_to_course_instance`.
+  // Be careful what you use it for.
+  const hasInstructorAccess = usernameNav.dataset.hasInstructorAccess === 'true';
+
+  document.querySelector('#navbar-load-from-disk')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_uid', { path: '/' });
+    Cookies.remove('pl_requested_course_role', { path: '/' });
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.remove('pl_requested_mode', { path: '/' });
+    Cookies.remove('pl_requested_date', { path: '/' });
+    Cookies.set('pl_requested_data_changed', { path: '/' });
```

Comment:
TypeScript has very helpfully pointed out that we're calling this function incorrectly; the second argument should be a cookie value string. In practice, this has sort of magically worked because the cookie has ended up with a value of `[Object object]`, which we interpret as truthy. I'm going to fix this in an upcoming commit, as the third options argument will be important when we need to explicitly specify a domain.

Reply:
Fixed in 48dcc13.